### PR TITLE
4631: added support for CatalogueSolutionSection setting on order sum…

### PR DESCRIPTION
**Pull request by SimonLLawrence**
…mary get method task 7623
Story [AB#4631](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/4631)
Task  [AB#7623](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7623)

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrdersController.cs [hunk 1]
```diff
@@ -1,136 +1,136 @@
...
-                createOrderResponse.OrderId = result.Value;
-                return CreatedAtAction(nameof(CreateOrderAsync).TrimAsync(), null, new { orderId = result.Value }, createOrderResponse);
-            }
-
-            createOrderResponse.Errors = result.Errors.Select(x => new ErrorModel(x.Id, x.Field));
-            return BadRequest(createOrderResponse);
-        }
-    }
-}
+﻿using System;
```
Comment (sam-roberts-bjss) [inline src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrdersController.cs:1]: On Github comparison it looks like the whole file has changed. Please ensure the file contains the expected line endings.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature [hunk 2]
```diff
@@ -1,152 +1,174 @@
...
+
+@5321
+Scenario: 1. Get the order summary
+    When the user makes a request to retrieve the order summary with the ID C000014-01
+    Then a response with status code 200 is returned
+    And the order summary is returned with the following values
+        | OrderId    | OrganisationId                       | Description   |
+        | C000014-01 | 4af62b99-638c-4247-875e-965239cd0c48 | A Description |
+    And the order Summary Sections have the following values
+        | Id                   | Status     |
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature:20]: There is probably a pending conflict with this file as I recently added the count column to each of the tests.

File: src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrdersController.cs (hunk 1 above)
Comment (SimonLLawrence) [reply src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrdersController.cs:1]: The file was CRLF in editor and i thought we were moving to LF, so i made the change?

Comment (SimonLLawrence) [reply src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrdersController.cs]: Changed it back to CRLF

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature (hunk 2 above)
Comment (SimonLLawrence) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary.feature]: Meged in development

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Extensions/OrderExtensionsTests.cs [hunk 3]
```diff
@@ -90,5 +90,28 @@ public void IsServiceRecipientsSectionComplete_ServiceRecipientsViewedFalse_Retu
...
+
+        [Test]
+        public void IsCatalogueSolutionsSectionComplete_NullOrder_ReturnsFalse()
+        {
+            var actual = OrderExtensions.IsCatalogueSolutionsSectionComplete(null);
+            actual.Should().BeFalse();
+        }
+
+        [Test]
+        public void IsCatalogueSolutionsSectionComplete_ServiceRecipientsViewed_ReturnsTrue()
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Extensions/OrderExtensionsTests.cs:102]: Please check the method name... `ServiceRecipientsViewed` should be `CatalogueSolutionsViewed`?

**Pull request merged**